### PR TITLE
Upgrade linter version

### PR DESCRIPTION
**Pull request by mortent**

### CODE REVIEW

File: internal/util/update/update_test.go [hunk 1]
```diff
@@ -309,7 +309,6 @@ func TestCommand_Run_localPackageChanges(t *testing.T) {
 				// Update upstream to Dataset2
 				UpstreamChanges: []Content{{Data: testutil.Dataset2}},
 			}
-			//defer g.Clean()
```
Comment (phanimarupaka) [inline internal/util/update/update_test.go:312]: This line should not be commented in first place. Must be a typo. Please add it back.

File: internal/cmdexport/cmdexport_test.go [hunk 2]
```diff
@@ -353,7 +353,7 @@ func (t *TestCase) ReplaceDIRMacro() error {
 
 	var params []string
 	for _, param := range t.params {
-		param = strings.Replace(param, "{DIR}", cwd, -1)
+		param = strings.Replace(param, "{DIR}", cwd, -1) //nolint: gocritic
```
Comment (phanimarupaka) [inline internal/cmdexport/cmdexport_test.go:356]: Why is this?

File: release/formula/main.go [hunk 3]
```diff
@@ -72,7 +72,7 @@ func getSha(version string) string {
 		"https://github.com/GoogleContainerTools/kpt/archive/" + version + ".tar.gz")
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "%v", err)
-		os.Exit(1)
+		os.Exit(1) //nolint:gocritic
```
Comment (phanimarupaka) [inline release/formula/main.go:75]: Why?

File: Makefile [hunk 4]
```diff
@@ -57,7 +57,7 @@ license:
 	GOBIN=$(GOBIN) scripts/update-license.sh
 
 lint:
-	(which golangci-lint || go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.22.2)
+	(which golangci-lint || go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.31.0)
```
Comment (phanimarupaka) [inline Makefile:60]: Why not get the latest always ?

Comment (mortent) [reply Makefile:60]: Upgrading the linter can lead to new lint errors. We should avoid random PRs getting build errors.

Comment (phanimarupaka) [reply Makefile:60]: Okay

**Review by phanimarupaka — APPROVED** [review]

**Pull request merged**